lib/s3routes/routes/routePOST.ts

```diff
@@ -58,6 +58,10 @@ export default function routePOST(
                     corsHeaders));
     }
 
+    if (objectKey === undefined && Object.keys(query).length === 0) {
```

Comment:
also on post object aws returns a 400 instead of a 501 Not Implemented for query string

```
    <Code>InvalidArgument</Code>
    <Message>Query String Parameters not allowed on POST requests.</Message>
```

Reply:
e109d2dcfffd083bf35c0e3db2715484ffab8efd
Tested here
0d40c42b4ba4d847dc1cace636c2349ccea9e368

Reply:
Also before that if there is an `objectKey` and no query string match the previous conditions, aws returns 405 Method Not Allowed instead of 501 Not Implemented

```xml
    <Code>MethodNotAllowed</Code>
    <Message>The specified method is not allowed against this resource.</Message>
    <Method>POST</Method>
    <ResourceType>OBJECT</ResourceType>
```

Reply:
Like this?
4ef5748c028619edff10d6d38b21df43c8d63d88 

Tested
https://github.com/scality/cloudserver/pull/5601/commits/29ebf3f0eb702a15067c67656f78b3421d30eb9e